lib/mozart_fetcher/component.ex

```diff
@@ -26,8 +26,15 @@ defmodule MozartFetcher.Component do
          config = %Config{endpoint: endpoint},
          {:ok, %HTTPoison.Response{status_code: 200, body: body}}
        ) do
-    metric(config.id, endpoint, 200)
-    %{index: component_index, id: config.id, status: 200, envelope: Envelope.build(body)}
+    case Jason.decode(body, keys: :atoms) do
+      {:ok, data} ->
+        metric(config.id, endpoint, 200)
+        %{index: component_index, id: config.id, status: 200, envelope: data}
+
+      {:error, _err} ->
+        metric(config.id, endpoint, :json_decode_error)
+        %{index: component_index, id: config.id, status: 200, envelope: %{}}
```

Comment:
Previously we returned an empty envelope so the necessary keys were present. If it's an empty map would that cause issues if the keys do not exist? I know assembler validates against the keys, but wondering if an empty envelope would have hidden optional components not being fetched whereas we would now error due to the keys not existing.

Reply:
The keys not existing doesn't cause an error anywhere in assembler, although it does cause the component to be marked as not valid as this line will return false: https://github.com/bbc/mozart-assembler/blob/master/src/lib/mozart_assembler/validators/envelope.rb#L17

If anything I think this is safer. Currently if a component endpoint was responding with a 200 and some invalid JSON then fetcher would return an empty envelope with the correct keys and a 200 would be considered "valid" in assembler and therefore the empty content merged into the template. For non-must-succeed components this makes no difference, if an empty content is merged in, or an invalid component is left out equals the same behaviour. In the case of must-succeed components though this new behaviour is preferable I think, as before the page would be rendered but with blank content taking the place of the must-succeed components, whereas now it will correctly 500.